ts-client/src/ValorClient.ts

```diff
@@ -0,0 +1,334 @@
+import axios, { AxiosInstance } from 'axios';
+
+type Dataset = {
+  id: number;
+  name: string;
+  metadata: object;
```

Comment:
metadata is an undefiend object of some structure?

i dont think you're using metadata anywhere in Valor but `Partial<Record<string, any>>` might be a safer type here just in case future users do

Reply:
metadata is very generic, i'll switch to that type

Reply:
`Record<string, any>` might be good here because we always expect metadata to come back, it may or may not be an empty object?